docs/writing-rules/glossary.md

```diff
@@ -0,0 +1,152 @@
+---
+slug: glossary
+title: SAST and rule-writing glossary
+hide_title: true
+description: Definitions of static analysis and Semgrep rule-writing terms.
+tags:
+  - Rule writing
+---
+
+# Static analysis and rule-writing glossary
+
+The definitions provided here are specific to Semgrep.
+
+## Constant propagation
+
+<!-- Refers to state of a variable remaining constant throughout the program. Semgrep can analyze whether a variable carries a constant value at a given point. Both Semgrep OSS and Semgrep Pro Engine perform this analysis, with Semgrep Pro able to track the propagation across files. -->
+
+Constant propagation is a type of analysis where values known to be constant are substituted in later uses, such as detecting matches. Semgrep can perform constant propagation across files, unless you are running Semgrep OSS, which can only propagate within a file.
```

Comment:
```suggestion
Constant propagation is a type of analysis where values known to be constant are substituted in later uses, allowing the value to be used to detect matches. Semgrep can perform constant propagation across files, unless you are running Semgrep OSS, which can only propagate within a file.
```

Tiiiiny nit the word "use" is actually referring to the use in the code. For instance, in the example given, on line 30 `x` is declared, and on line 32 `x` is used. ("declared" and "used" are the technical PL terms, which is why this caught my attention.) Personally I think we can also just eliminate the clause after "later uses", because the example conveys it more clearly.